src/core/object/timer.ts

```diff
@@ -78,12 +82,15 @@ export default class Timer {
 	 *
 	 * @param seconds - Seconds
 	 */
-	update(seconds: number | null): void {
+	update(seconds: number | null | undefined): void {
 		// only if timer was started
 		if (this.startedOn !== null) {
 			this.targetSeconds = seconds;
 
-			if (this.targetSeconds !== null) {
+			if (
```

Comment:
won't this change mean that when there is no duration we would just fall into clearTrigger?

Reply:
Yes. However, crucially, by this point getSecondsToScrobble has been called.
getSecondsToScrobble sets the seconds to default scrobble time if it is null.
So really, it should be impossible for this to actually be null, just typescript type checking combined with javascript jank moment.